content/en/blog/_posts/2020-03-30-two-phased-canary-rollout-with-gloo.md

```diff
@@ -0,0 +1,772 @@
+              - name: stage
+                value: canary
+            prefix: /
+        routeAction:
+          single:
+            upstream:
+              name: echo
+              namespace: gloo-system
+            subset:
+              values:
+                version: v2
+      - matchers:
+          - prefix: /
+        routeAction:
+          multi:
+            destinations:
+              - destination:
+                  upstream:
+                    name: echo
+                    namespace: gloo-system
+                  subset:
+                    values:
+                      version: v1
+                # No traffic will be sent to v1 anymore
+                weight: 0
+              - destination:
+                  upstream:
+                    name: echo
+                    namespace: gloo-system
+                  subset:
+                    values:
+                      version: v2
+                # Now all the traffic will be routed to v2
+                weight: 100
+```
+
+We can apply that to the cluster with the following command: 
+```bash
+kubectl apply -f https://raw.githubusercontent.com/solo-io/gloo-ref-arch/blog-30-mar-20/platform/prog-delivery/two-phased-with-os-gloo/3-progressive-traffic-shift-to-v2/vs-3.yaml
+```
+
+Now when we send traffic to the gateway, we should see all of the requests return `version:v2`. 
+
+```bash
+➜ curl $(glooctl proxy url)/
+version:v2
+➜ curl $(glooctl proxy url)/
+version:v2
+➜ curl $(glooctl proxy url)/
+version:v2
+```
+
+### Decommissioning v1
+   
+At this point, we have deployed the new version of our application, conducted correctness tests using subset routing, 
+conducted load and performance tests by progressively shifting traffic to the new version, and finished 
+the rollout. The only remaining task is to clean up our `v1` resources. 
+
+First, we'll clean up our routes. We'll leave the subset specified on the route so we are all setup for future upgrades. 
+
+```yaml
+apiVersion: gateway.solo.io/v1
+kind: VirtualService
+metadata:
+  name: echo
+  namespace: gloo-system
+spec:
+  virtualHost:
+    domains:
+      - '*'
+    routes:
+      - matchers:
+          - prefix: /
+        routeAction:
+          single:
+            upstream:
+              name: echo
+              namespace: gloo-system
+            subset:
+              values:
+                version: v2
+```
+
+We can apply this update with the following command: 
+
+```bash
+kubectl apply -f https://raw.githubusercontent.com/solo-io/gloo-ref-arch/blog-30-mar-20/platform/prog-delivery/two-phased-with-os-gloo/4-decommissioning-v1/vs.yaml
+```
+
+And we can delete the `v1` deployment, which is no longer serving any traffic. 
+            
+```bash
+kubectl delete deploy -n echo echo-v1
+```
+
+Now our cluster looks like this:
+
+![](https://raw.githubusercontent.com/solo-io/gloo-ref-arch/blog-30-mar-20/platform/prog-delivery/two-phased-with-os-gloo/4-decommissioning-v1/end-state.png)
```

Comment:
1. This image should have an alt attribute.
2. Can you copy the image into the repo so that it's served via the content CDN and the image is in source control?